### PR TITLE
 Bug 1731263: Re-enable preemption test with fixes

### PR DESCRIPTION
**Pull request by damemi**
Rebased changes from https://github.com/openshift/origin/pull/23050

### CODE REVIEW

File: vendor/k8s.io/kubernetes/test/e2e/scheduling/preemption.go [hunk 1]
```diff
@@ -242,17 +303,18 @@ var _ = SIGDescribe("SchedulerPreemption [Serial]", func() {
...
-			currentNode, err := cs.CoreV1().Nodes().Get(node.Name, metav1.GetOptions{})
-			framework.ExpectNoError(err)
-			cpuAllocatable, found := currentNode.Status.Allocatable["cpu"]
+			cpuAllocatable, found := node.Status.Allocatable["cpu"]
 			Expect(found).To(Equal(true))
 			milliCPU := cpuAllocatable.MilliValue()
-			milliCPU = milliCPU * 10 / 100
-			memAllocatable, found := currentNode.Status.Allocatable["memory"]
+			// Just to be tolerant use 0.6 of resources available on the node
+			milliCPU = int64(float64(milliCPU-currentMemUsage) * float64(0.6))
```
Comment (soltysh) [inline vendor/k8s.io/kubernetes/test/e2e/scheduling/preemption.go]: @damemi it looks like this ends up being a negative number in the tests: `Pod \"pod0-sched-preemption-medium-priority\" is invalid: spec.containers[0].resources.requests[cpu]: Invalid value: \"-912259020m\": must be greater than or equal to 0` maybe you'd need similar checks for utilization like in the other test?

File: vendor/k8s.io/kubernetes/test/e2e/scheduling/preemption.go [hunk 2]
```diff
@@ -90,45 +90,67 @@ var _ = SIGDescribe("SchedulerPreemption [Serial]", func() {
...
+			// Just to be tolerant use 0.6 of resources available on the node
+			memory = int64(float64(memory-currentMemUsage) * float64(0.6))
 			podRes = v1.ResourceList{}
+			// If a node is already heavily utilized let not's create a pod there.
+			if milliCPU <= 0 {
+				framework.Logf("Node is heavily utilized, let's not create a pod here")
+				continue
+			}
 			podRes[v1.ResourceCPU] = *resource.NewMilliQuantity(int64(milliCPU), resource.DecimalSI)
-			podRes[v1.ResourceMemory] = *resource.NewQuantity(int64(memory), resource.BinarySI)
```
Comment (damemi) [inline vendor/k8s.io/kubernetes/test/e2e/scheduling/preemption.go]: I'm noticing this now and it seems weird, @ravisantoshgudimetla do you know why we wouldn't want to set the `resourceMemory`, especially after we calculated it above? Or could this have been an error?

File: vendor/k8s.io/kubernetes/test/e2e/scheduling/preemption.go [hunk 3]
```diff
@@ -156,46 +186,68 @@ var _ = SIGDescribe("SchedulerPreemption [Serial]", func() {
...
+			// Just to be tolerant use 0.6 of resources available on the node
+			memory = int64(float64(memory-currentMemUsage) * float64(0.6))
 			podRes = v1.ResourceList{}
+			// If a node is already heavily utilized let not's create a pod there.
+			if milliCPU <= 0 {
+				framework.Logf("Node is heavily utilized, let's not create a pod there")
+				continue
+			}
 			podRes[v1.ResourceCPU] = *resource.NewMilliQuantity(int64(milliCPU), resource.DecimalSI)
-			podRes[v1.ResourceMemory] = *resource.NewQuantity(int64(memory), resource.BinarySI)
```
Comment (damemi) [inline vendor/k8s.io/kubernetes/test/e2e/scheduling/preemption.go]: @ravisantoshgudimetla same question here

File: vendor/k8s.io/kubernetes/test/e2e/scheduling/preemption.go [hunk 4]
```diff
@@ -90,45 +90,67 @@ var _ = SIGDescribe("SchedulerPreemption [Serial]", func() {
 		var podRes v1.ResourceList
```
Comment (sttts) [inline vendor/k8s.io/kubernetes/test/e2e/scheduling/preemption.go]: can we please start using sensible commit messages? This one says nothing.

**Pull request merged**